src/pages/dev/general-message-passing/cosmos-gmp.mdx

```diff
@@ -164,4 +163,28 @@ pub fn multi_send_to_evm(
 ## Chain Support
 For a Cosmos blockchain to support receiving GMP messages (naively or via CosmWasm), it should implement the appropriate middleware in Go as part of the consensus layer.
 
-See our [Sample GMP Middleware](https://github.com/axelarnetwork/evm-cosmos-gmp-sample/tree/main/native-integration/sample-middleware).
\ No newline at end of file
+If you have CosmWasm on your Cosmos chain and want to be able to receieve and deliver messages, install [Osmosis IBC-hooks](https://github.com/osmosis-labs/osmosis/tree/main/x/ibc-hooks).
+
+If you don’t support CosmWasm on your Cosmos chain, you'll need to implement and install your own Axelar GMP module to receive and execute messages. See our [sample reference implementation](https://github.com/axelarnetwork/evm-cosmos-gmp-sample/tree/main/native-integration/sample-middleware).
```

Comment:
Just a question that comes to mind reading this part. Up to you if you think it's worth addressing this in the docs. 
Is there any difference in whether my chain allows for GMP via using CosmWasm on my chain vs using an Axelar GMP module? 
In other words is there an incentive for me to go one way or another in terms of limitations I might experience if for example, I integrate GMP via Axelar GMP module instead of CosmWasm

Reply:
This isn't decided by us. Having smart contract support on your chain is determined by a lot of external factors, and folks aren't adding CosmWasm support just to get Axelar's GMP, at least not today.